lib/std/testing.zig

```diff
@@ -191,6 +191,14 @@ pub fn expect(ok: bool) void {
     if (!ok) @panic("test failure");
 }
 
+/// This function is intended to be used only in tests. When the two strings are not
+/// equal, prints diagnostics to stderr to show exactly how they are not equal, then aborts.
+pub fn expectEqualString(expected: []const u8, actual: []const u8) void {
```

Comment:
`u16` can be useful For Windows.

Reply:
so either I can add `expectEqualUtf8` and `expectEqualUtf16` (I assume UTF-8 strings are encoded as `[]const u8` and UTF-16 strings are encoded as `|]const u16`) or I can add a type parameter to `expectEqualString`.

I'm not sure which is the best.

Reply:
Or using `Utf8Iterator` and `Utf16LeIterator` from `std.unicode` codepoint by codepoint.
And something like `pub fn expectEqualString(expected: var, actual: @TypeOf(expected))`

Reply:
I'm not familiar with those struct yet, I'll take a look

Reply:
I have something that works using `Utf8Iterator` and `Utf16LeIterator` but it's significantly more complex now. The diagnostic when the utf16 strings are differnt is also not helpful since `std.debug.panic` can't print the utf16 slice as a string, at which point you're better served just using `expectEqualSlices` I think ?

Given the goal for this function is to have better diagnostics when comparing strings I'm not sure having utf16 support is that useful.

Reply:
What about this:
```zig
pub fn expectEqualString(comptime T: type, expected: []const T, actual: []const T) void {
    if (!std.mem.eql(T, expected, actual)) {
        std.debug.panic("expected string \"{}\", got \"{}\"", .{ expected, actual });
    }
}
```

Reply:
yeah I'm fine with that. but printing a `[]const u16` on my Linux box outputs `u16@202490` which is not that useful. Maybe it's different on Windows ?

Reply:
I pushed an update using that signature. Still not sure if it prints the u16 string correctly on Windows though.